scripts/check-go-version.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+# checks if the installed Go version is less than the required version.
+# returns a warning message if it's less than required. Otherwise, returns "Building Conduit..."
+
+function version_lt() { test "$(echo "$@" | tr " " "\n" | sort -rV | head -n 1)" != "$1"; }
+
+GO_VERSION=$(go version | { read -r _ _ v _; echo "${v#go}"; })
+# needs to be changed each time conduit Go version is upgraded
+REQUIRED_GO_VERSION="1.18"
+
+if version_lt "$GO_VERSION" "$REQUIRED_GO_VERSION"; then
+    echo "WARNING: min Go version required is go$REQUIRED_GO_VERSION, version installed is go$GO_VERSION"
+else
+    echo "Building Conduit..."
```

Comment:
I would skip the `else` part to make the script a bit more general (we might use it in other places than building Conduit in the future).

Reply:
I added it so that it wouldn't just print an empty line before the build command.. it looked weird

Reply:
but yeah didn't mind it that much.. will take it out